storage/range.go

```diff
@@ -257,31 +258,37 @@ func (r *Range) IsLeader() bool {
 	return true
 }
 
+// Desc returns the range's descriptor. This method is thread-safe.
+func (r *Range) Desc() *proto.RangeDescriptor {
+	return (*proto.RangeDescriptor)(atomic.LoadPointer(&r.desc))
+}
+
+// SetDesc sets the range's descriptor. This method is thread-safe.
```

Comment:
Seems like there needs to be some higher level locking here. Add a comment that such locking already exists.